frontend/src/e2e-test/specs/7_messaging/messaging.spec.ts

```diff
@@ -664,6 +664,46 @@ describe('Sending and receiving messages', () => {
         )
       })
 
+      test('Citizen session is kept alive as long as user keeps typing', async () => {
+        await openSupervisorPage(mockedDateAt10)
+        await unitSupervisorPage.goto(`${config.employeeUrl}/messages`)
+        const messagesPage = new MessagesPage(unitSupervisorPage)
+        const messageEditor = await messagesPage.openMessageEditor()
+        await messageEditor.sendNewMessage(defaultMessage)
+        await runPendingAsyncJobs(mockedDateAt10.addMinutes(1))
+
+        await openCitizen(mockedDateAt11)
+
+        await citizenPage.page.setExtraHTTPHeaders({
+          'X-Session-TTL': '1000'
+        })
+        await citizenPage.goto(config.enduserMessagesUrl)
+        const citizenMessagesPage = new CitizenMessagesPage(citizenPage)
+        await citizenPage.page.evaluate(() => {
+          if (window.evaka) window.evaka.keep_session_alive_throttle_time = 300 // Set to 300ms for tests
+        })
+        await citizenMessagesPage.assertThreadContent(defaultMessage)
+
+        await citizenMessagesPage.startReplyToFirstThread()
+        const slowTypedText =
+          'Olen aika hidas kirjoittamaan näitä viestejä, mutta yritän parhaani: Lapseni ovat sitä ja tätä...'
+        // typing this takes 2x longer than the session TTL
+        await citizenMessagesPage
```

Comment:
Voisiko tässä headerien pyörittelyn sijaan katsoa että sessiokeksin expires on siirtynyt vähintään tietyn määrän? Silloin ei tarvisi apigw:ssä headereihin liittyvää testikoodia joka on nyt myös tuotannossa

```ts
const sessionExpires = (await page.page.context().cookies()).find(({name}) => name === 'evaka.eugw.session')?.expires
```

Reply:
Ehkä näitä varten voisi olla ihan `getCitizenSessionCookie(page)` ja `clearCitizenSessionCookie(page)` -funktiot niin itse testi pysyisi siistinä

Reply:
muutettu expires tarkastukseksi